enterprise/api/v1/views.py

```diff
@@ -334,10 +351,11 @@ def enroll_learners_in_courses(self, request, pk):
             except LinkUserToEnterpriseError:
                 email_errors.append(email)
 
-        # Remove the bad emails from enrollments_info and the emails set, don't attempt to enroll or link bad emails.
-        for errored_user in email_errors:
-            enrollments_info[:] = [info for info in enrollments_info if info['email'] != errored_user]
-            emails.remove(errored_user)
+        # Remove the bad emails and bad user_ids from enrollments_info; don't attempt to enroll or link them.
+        enrollments_info = [
+            info for info in enrollments_info
+            if 'email' in info and info['email'] not in email_errors and info.get('user_id') not in user_id_errors
```

Comment:
could you explain why we're checking for email and user id exists in the errors lists, in different ways? Is there a reason why we couldn't do `info.get('email')` or `'user_id' in info and info['user_id']` to keep it consistent?

Reply:
I deliberated both ways, in the end I thought it was more robust to be more opinionated about there always being an email.  This is not a strongly held opinion and can go either way---the way you described is probably more readable if you're asking about it